TypeScript error with `defineSlots` and generic type since vue-tsc 2.2.4

### Vue - Official extension or vue-tsc version

2.2.4, 2.2.6

### VSCode version

1.97.2

### Vue version

3.5.13

### TypeScript version

5.8.2

### System Info

```shell
System:
    OS: macOS 15.1.1
    CPU: (8) arm64 Apple M2
    Memory: 681.22 MB / 24.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.12.0 - ~/.volta/tools/image/node/22.12.0/bin/node
    npm: 11.1.0 - ~/.volta/tools/image/npm/11.1.0/bin/npm
    bun: 1.0.4 - ~/.bun/bin/bun
  Browsers:
    Brave Browser: 133.1.75.181
    Chrome: 131.0.6778.140
    Safari: 18.1.1
```

### package.json dependencies

```json
"devDependencies":
{
    "typescript": "^5.8.2",
    "vue": "^3.5.13",
    "vue-tsc": "2.2.4"
}
```

### Steps to reproduce

```vue
<script generic="T extends string" lang="ts" setup>
interface Props
{
    slots: T[];
}
defineProps<Props>();
defineSlots<{[K in `slot:${T}`]?: () => any}>();
</script>
<template>
    <div v-for="slot of slots">
        <slot v-bind:name="`slot:${slot}`"/>
    </div>
</template>
```

When running `npx vue-tsc --noEmit`:
- With vue-tsc 2.2.2 no error is reported
- With vue-tsc 2.2.4 and vue-tsc 2.2.6 the following error is reported:

    ```
    Example.vue:11:10 - error TS2345: Argument of type '{}' is not assignable to parameter of type 'NonNullable<(Readonly<__VLS_Slots> & __VLS_Slots)[`slot:${T}`]> extends (props: infer P) => any ? P : {}'.

    11         <slot v-bind:name="`slot:${slot}`"/>
    ```

### What is expected?

No error.

### What is actually happening?

```
Example.vue:11:10 - error TS2345: Argument of type '{}' is not assignable to parameter of type 'NonNullable<(Readonly<__VLS_Slots> & __VLS_Slots)[`slot:${T}`]> extends (props: infer P) => any ? P : {}'.

11         <slot v-bind:name="`slot:${slot}`"/>
```

### Link to minimal reproduction

_No response_

### Any additional comments?

_No response_

--- a/packages/language-core/lib/codegen/globalTypes.ts
+++ b/packages/language-core/lib/codegen/globalTypes.ts
@@ -158,7 +158,7 @@ export function generateGlobalTypes({
 		: (_: {}${checkUnknownProps ? '' : ' & Record<string, unknown>'}, ctx?: any) => { __ctx?: { attrs?: any, expose?: any, slots?: any, emit?: any, props?: {}${checkUnknownProps ? '' : ' & Record<string, unknown>'} } };
 	function __VLS_functionalComponentArgsRest<T extends (...args: any) => any>(t: T): 2 extends Parameters<T>['length'] ? [any] : [];
 	function __VLS_asFunctionalElement<T>(tag: T, endTag?: T): (attrs: T${checkUnknownComponents ? '' : ' & Record<string, unknown>'}) => void;
-	function __VLS_asFunctionalSlot<S>(slot: S): (props: NonNullable<S> extends (props: infer P) => any ? P : {}) => void;
+	function __VLS_asFunctionalSlot<S>(slot: S): S extends () => infer R ? (props: {}) => R : NonNullable<S>;
 	function __VLS_tryAsConstant<const T>(t: T): T;
 }
 `;

--- a/test-workspace/tsc/passedFixtures/vue3/#5228/main.vue
+++ b/test-workspace/tsc/passedFixtures/vue3/#5228/main.vue
@@ -1,0 +1,13 @@
+<script setup lang="ts" generic="T extends string">
+defineSlots<{
+	[K in `slot:${T}`]?: () => any;
+}>();
+
+let slots!: T[];
+</script>
+
+<template>
+	<div v-for="slot of slots">
+		<slot :name="`slot:${slot}`"></slot>
+	</div>
+</template>

